parsec/core/gui/parsec_application.py

```diff
@@ -2,29 +2,32 @@
 from __future__ import annotations
 
 import sys
+from typing import Set, Tuple
 
-from PyQt5.QtWidgets import QApplication
+from PyQt5.QtWidgets import QApplication, QWidget
 from PyQt5.QtCore import QFile, QEvent
 from PyQt5.QtGui import QFont, QFontDatabase, QPalette, QColor
 from structlog import get_logger
 
+from parsec._parsec import DeviceID, OrganizationID
+
 logger = get_logger()
 
 
 class ParsecApp(QApplication):
-    connected_devices = set()
+    connected_devices: Set[Tuple[OrganizationID, DeviceID]] = set()
 
-    def __init__(self):
+    def __init__(self) -> None:
         super().__init__(["-stylesheet"])
         self.setOrganizationName("Scille")
         self.setOrganizationDomain("parsec.cloud")
         self.setApplicationName("Parsec")
-        pal = self.palette()
+        pal = self.palette()  # type: ignore[misc]
```

Comment:
Why the `ignore`? It should just be a `QPalette` or am I missing something?

Reply:
`self` doesn't have `palette` method

Reply:
But it inherits `QApplication` which inherits `QGuiApplication` which does have a `palette()` method. Is it a bug in PyQt typing, or mypy does not take inheritance into account? Is it possible to cast `self` into a `QGuiApplication` to avoid the  `ignore`?